sled-agent/src/bootstrap/early_networking.rs

```diff
@@ -905,186 +896,246 @@ impl From<EarlyNetworkConfig> for bootstore::NetworkConfig {
     }
 }
 
-#[derive(Clone, Debug, Deserialize, Serialize, JsonSchema)]
-struct EarlyNetworkConfigBodyV1 {
-    /// The external NTP server addresses.
-    pub ntp_servers: Vec<String>,
+/// Structures and routines used to maintain backwards compatibility.  The
+/// contents of this module should only be used to convert older data into the
+/// current format, and not for any ongoing run-time operations.
+pub mod back_compat {
```

Comment:
Good idea to throw this into a separate module.

(For any other reviewers, the contents of this file appear to mostly be re-shuffling these structs, rather than changing them significantly)